model/pipeline/src/main/proto/beam_runner_api.proto

```diff
@@ -576,7 +581,12 @@ service TestStreamService {
   // A TestStream will request for events using this RPC.
   rpc Events(EventsRequest) returns (stream TestStreamPayload.Event) {}
 }
-message EventsRequest {}
+
+message EventsRequest {
+  // The set of keys to read from. The keys are the specific PCollections to
+  // read.
+  repeated string keys = 1;
```

Comment:
Can you provide more context as to why this is needed?

Is this the pcollection id, or the output name or the ptransform id representing the test stream?

Reply:
Changed to comment to: The set of tags to read from. These tags are a subset of the TestStreamPayload's output_tags. This allows Interactive Beam to cache many PCollections from a pipeline then replay a subset of them.